tracer/src/Datadog.Trace/IAST/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -0,0 +1,214 @@
+        res.Add(new SecretRegex("shopify-access-token", @"shpat_[a-fA-F0-9]{32}"));
+        res.Add(new SecretRegex("shopify-custom-access-token", @"shpca_[a-fA-F0-9]{32}"));
+        res.Add(new SecretRegex("shopify-private-app-access-token", @"shppa_[a-fA-F0-9]{32}"));
+        res.Add(new SecretRegex("sidekiq-sensitive-url", @"(?i)\b(http(?:s??):\/\/)([a-f0-9]{8}:[a-f0-9]{8})@(?:gems.contribsys.com|enterprise.contribsys.com)(?:[\/|\#|\?|:]|$)"));
+        res.Add(new SecretRegex("slack-bot-token", @"(xoxb-[0-9]{10,13}\-[0-9]{10,13}[a-zA-Z0-9-]*)"));
+        res.Add(new SecretRegex("slack-user-token", @"(xox[pe](?:-[0-9]{10,13}){3}-[a-zA-Z0-9-]{28,34})"));
+        res.Add(new SecretRegex("slack-app-token", @"(?i)(xapp-\d-[A-Z0-9]+-\d+-[a-z0-9]+)"));
+        res.Add(new SecretRegex("slack-config-access-token", @"(?i)(xoxe.xox[bp]-\d-[A-Z0-9]{163,166})"));
+        res.Add(new SecretRegex("slack-config-refresh-token", @"(?i)(xoxe-\d-[A-Z0-9]{146})"));
+        res.Add(new SecretRegex("slack-legacy-bot-token", @"(xoxb-[0-9]{8,14}\-[a-zA-Z0-9]{18,26})"));
+        res.Add(new SecretRegex("slack-legacy-workspace-token", @"(xox[ar]-(?:\d-)?[0-9a-zA-Z]{8,48})"));
+        res.Add(new SecretRegex("slack-legacy-token", @"(xox[os]-\d+-\d+-\d+-[a-fA-F\d]+)"));
+        res.Add(new SecretRegex("slack-webhook-url", @"(https?:\/\/)?hooks.slack.com\/(services|workflows)\/[A-Za-z0-9+\/]{43,46}"));
+        res.Add(new SecretRegex("square-access-token", @"(?i)\b(sq0atp-[0-9A-Za-z\-_]{22})(?:['|""|\n|\r|\s|\x60|;]|$)"));
+        res.Add(new SecretRegex("square-secret", @"(?i)\b(sq0csp-[0-9A-Za-z\-_]{43})(?:['|""|\n|\r|\s|\x60|;]|$)"));
+        res.Add(new SecretRegex("stripe-access-token", @"(?i)(sk|pk)_(test|live)_[0-9a-z]{10,32}"));
+        res.Add(new SecretRegex("microsoft-teams-webhook", @"https:\/\/[a-z0-9]+\.webhook\.office\.com\/webhookb2\/[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}@[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}\/IncomingWebhook\/[a-z0-9]{32}\/[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}"));
+        res.Add(new SecretRegex("telegram-bot-api-token", @"(?i)(?:^|[^0-9])([0-9]{5,16}:A[a-zA-Z0-9_\-]{34})(?:$|[^a-zA-Z0-9_\-])"));
+        res.Add(new SecretRegex("twilio-api-key", @"SK[0-9a-fA-F]{32}"));
+        res.Add(new SecretRegex("vault-service-token", @"(?i)\b(hvs\.[a-z0-9_-]{90,100})(?:['|""|\n|\r|\s|\x60|;]|$)"));
+        res.Add(new SecretRegex("vault-batch-token", @"(?i)\b(hvb\.[a-z0-9_-]{138,212})(?:['|""|\n|\r|\s|\x60|;]|$)"));
+
+        return res;
+    }
+
+    private void RunShutdown()
+    {
+        try
+        {
+            _started = false;
+            _waitEvent.Set();
+        }
+        catch { }
+    }
+
+    private struct SecretRegex
+    {
+        public string Rule;
+        public Regex Regex;
+
+        public SecretRegex(string rule, string regex)
+        {
+            Rule = rule;
+            Regex = new Regex(regex, RegexOptions.Compiled);
+        }
+    }
```

Comment:
Can we make this a `readonly` struct?
```suggestion
    private readonly struct SecretRegex
    {
        public readonly string Rule;
        public readonly Regex Regex;

        public SecretRegex(string rule, string regex)
        {
            Rule = rule;
            Regex = new Regex(regex, RegexOptions.Compiled);
        }
    }
```

Reply:
Of course. On it